src/software/sensor_fusion/sensor_fusion_test.cpp

```diff
@@ -1010,3 +1010,289 @@ TEST_F(SensorFusionTest, breakbeam_fail_test_ssl)
     // did it not use robot position
     EXPECT_TRUE(ball_position != robot_state.position());
 }
+
+TEST_F(SensorFusionTest, test_detect_injured_robots_with_no_error)
+{
+    /* create a list of robot status messages */
+    std::vector<TbotsProto::RobotStatus> robot_status_msgs;
+    for (int i = 0; i < 3; i++)
+    {
+        auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
+        robot_msg->set_robot_id(i);
+        robot_msg->add_error_code(TbotsProto::NO_ERROR);
+        robot_status_msgs.push_back(*robot_msg);
+    }
+
+    SensorProto sensor_msg;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    /* call detect error on a list of normal robots */
+    sensor_fusion.detectInjuredRobots(robot_status_msgs);
+
+    /* test to see if any robots gets added to injured_robots */
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 0);
+}
+
+TEST_F(SensorFusionTest, test_detect_one_injured_robot_with_high_cap)
+{
+    std::vector<TbotsProto::RobotStatus> robot_status_msgs;
+    auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg->set_robot_id(1);
+    robot_msg->add_error_code(TbotsProto::ErrorCode::HIGH_CAP);
+    robot_status_msgs.push_back(*robot_msg);
+
+    auto robot_msg2 = std::make_unique<TbotsProto::RobotStatus>();
+    robot_msg2->add_error_code(TbotsProto::ErrorCode::NO_ERROR);
+    robot_msg2->set_robot_id(2);
+    robot_status_msgs.push_back(*robot_msg2);
+
+    SensorProto sensor_msg;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.processSensorProto(sensor_msg);
+
+    sensor_fusion.detectInjuredRobots(robot_status_msgs);
+
+    std::vector<Robot> injured_robots =
+        sensor_fusion.getWorld().value().friendlyTeam().getInjuredRobots();
+    EXPECT_EQ(injured_robots.size(), 1);
```

Comment:
kinda pedantic but if you use `ASSERT_EQ` here, then below you can remove the `if` because the assertion will stop execution at this point if it fails